### ISSUE TITLE
Autonomous research loop for TerrascryUNet architecture and hyperparameter search

### ISSUE THREAD

**Issue by chadlindell** [chadlindell/terrascry#590]
## Context

Inspired by [karpathy/autoresearch](https://github.com/karpathy/autoresearch) — an autonomous AI agent loop that iteratively modifies a training script, runs fixed-budget experiments, keeps improvements, discards regressions, and repeats indefinitely.

The pattern maps directly onto our ML training pipeline:

| autoresearch | TERRASCRY |
|---|---|
| `train.py` (GPT model) | `geosim/training/models/unet.py` (TerrascryUNet) |
| `prepare.py` (data/eval) | `geosim/training/chip_extractor.py` + `benchmark.py` |
| `program.md` (agent instructions) | New `RESEARCH.md` with geophysics constraints |
| val_bpb metric | Composite score from benchmark suite (#536) |
| 5-min time budget | Configurable per-experiment budget |

### `geosim/training/autoresearch/`

1. **`program.md`** — Agent instructions for TerrascryUNet experiments:
   - Domain constraints (physics regularizer must be preserved, GroupNorm for batch-size independence, <2M params, ONNX-exportable)
   - What's fair game: encoder widths, skip connection types, activation functions, head designs, loss weights, augmentation, lr/batch_size
   - Evaluation protocol: fixed synthetic pre-train checkpoint → fine-tune → benchmark suite composite score
   - Simplicity criterion: simpler is better at equal performance

2. **`run_experiment.py`** — Fixed-budget training + evaluation:
   - Load frozen pre-train checkpoint (or train from scratch with budget)
   - Run fine-tune stage
   - Execute benchmark suite (Pd@Pfa, MAE, SSIM, injection-recovery)
   - Compute composite scalar score for keep/discard decision
   - Print summary in grep-friendly format

3. **`results.tsv`** — Experiment log (commit, score, memory, status, description)

4. **`autoresearch.py`** — Orchestrator script (optional, for non-agent use):
   - Loop: modify → train → evaluate → keep/discard → repeat
   - Git branch per research run (`autoresearch/{tag}`)

### Composite scoring function

Weighted combination of benchmark metrics into a single scalar:
```
score = w_pd * Pd@Pfa=0.05 + w_mae * (1 - norm_MAE) + w_ssim * SSIM + w_consistency * repeat_line_correlation
```

Weights TBD via AI counsel.

## Key differences from autoresearch

- **Multi-metric evaluation** — Not just a single loss; composite score from multiple geophysics-relevant metrics
- **Two-stage training** — Pre-train on synthetic → fine-tune on real; experiments can target either stage
- **Physics constraints** — Agent must preserve ∇·B=0 regularizer, coverage mask channel, and ONNX exportability
- **Domain-specific augmentation** — Heading-aware rotations only (not free rotation)

## Dependencies

- #539 (U-Net architecture) ✅
- #541 (physics loss) ✅
- #542 (training pipeline) ✅
- #536 (benchmark suite) ✅
- #543 (ONNX export) ✅

All dependencies are complete.

## Acceptance criteria

- [ ] Agent can run autonomously for 8+ hours without intervention
- [ ] Each experiment completes within configurable time budget
- [ ] Results logged to TSV with full git traceability
- [ ] Composite score correctly reflects geophysics-relevant performance
- [ ] Physics constraints enforced (agent cannot remove regularizer)
- [ ] `program.md` is sufficient for any Claude/Codex agent to operate
